File: frontend/src/lib/homepage.ts
export type HomepageCta = {
  label: string;
  href: string;
};

export type HomepageHeroConfig = {
  badge: string;
  title: string;
  subtitle: string;
  primary_cta: HomepageCta;
  secondary_cta: HomepageCta;
  mobile_categories_cta: HomepageCta;
  categories_link_label: string;
  categories_link_href: string;
  show_categories_bento: boolean;
  category_limit: number;
};

export type HomepageFeaturedConfig = {
  title: string;
  subtitle: string;
  catalog_label: string;
  catalog_href: string;
  product_count: number;
  parent_slug: string | null;
};

export type HomepageSection = {
  id: string;
  enabled: boolean;
};

export type HomepageConfig = {
  sections: HomepageSection[];
  hero: HomepageHeroConfig;
  featured: HomepageFeaturedConfig;
  show_promo_fallback: boolean;
};

export type HomepageAdminBundle = HomepageConfig & {
  banners_hero: import("@/lib/home-banners").HomeBannerAdmin[];
  banners_promo: import("@/lib/home-banners").HomeBannerAdmin[];
};

export const HOMEPAGE_SECTION_META: Record<
  string,
  { label: string; description: string; hint?: string }
> = {
  carousel: {
    label: "اسلایدر بنر",
    description: "بالای صفحه — تصاویر تمام‌عرض",
    hint: "حداقل یک بنر با محل «اسلایدر» اضافه کنید",
  },
  hero: {
    label: "معرفی و دسته‌ها",
    description: "عنوان، دکمه‌ها و شبکهٔ دسته‌بندی",
    hint: "دسته‌ها از بخش «دسته‌ها» در ادمین مدیریت می‌شوند",
  },
  featured: {
    label: "محصولات منتخب",
    description: "گرید محصولات پرفروش یا جدید",
  },
  promo: {
    label: "بنرهای تبلیغاتی",
    description: "کارت‌ها یا تصاویر پایین صفحه",
    hint: "اگر بنری نباشد، می‌توانید بخش‌های پیش‌فرض را فعال کنید",
  },
};

export const DEFAULT_HOMEPAGE_CONFIG: HomepageConfig = {
  sections: [
    { id: "carousel", enabled: true },
    { id: "hero", enabled: true },
    { id: "featured", enabled: true },
    { id: "promo", enabled: true },
  ],
  hero: {
    badge: "فروشگاه طرح‌محور",
    title: "دنیای خودت رو انتخاب کن",
    subtitle: "تیشرت و هودی با موضوع دلخواه — یا با Design Lab اثر خودت را بساز و سفارش بده.",
    primary_cta: { label: "شروع Design Lab", href: "/customize" },
    secondary_cta: { label: "ویترین خالقین", href: "/studios" },
    mobile_categories_cta: { label: "مرور دسته‌ها", href: "/browse" },
    categories_link_label: "همه دسته‌ها",
    categories_link_href: "/browse",
    show_categories_bento: true,
    category_limit: 6,
  },
  featured: {
    title: "منتخب",
    subtitle: "طرح‌های تازه",
    catalog_label: "کاتالوگ ←",
    catalog_href: "/catalog",
    product_count: 6,
    parent_slug: null,
  },
  show_promo_fallback: true,
};

export function isSectionEnabled(config: HomepageConfig, id: string) {
  return config.sections.find((s) => s.id === id)?.enabled ?? true;
}

File: backend/app/schemas/homepage.py
"""تنظیمات صفحهٔ اصلی فروشگاه."""

from pydantic import BaseModel, Field


class HomepageCta(BaseModel):
    label: str = ""
    href: str = ""


class HomepageHeroConfig(BaseModel):
    badge: str = "فروشگاه دشتستان"
    title: str = "لوازم خانگی و سبک زندگی"
    subtitle: str = "خرید آنلاین با گارانتی اصلی، قیمت رقابتی و ارسال سریع."
    primary_cta: HomepageCta = Field(default_factory=lambda: HomepageCta(label="مشاهده کاتالوگ", href="/catalog"))
    secondary_cta: HomepageCta = Field(default_factory=lambda: HomepageCta(label="سفارش سازمانی", href="/business"))
    mobile_categories_cta: HomepageCta = Field(default_factory=lambda: HomepageCta(label="مرور دسته‌ها", href="/browse"))
    categories_link_label: str = "همه دسته‌ها"
    categories_link_href: str = "/browse"
    show_categories_bento: bool = True
    category_limit: int = Field(default=6, ge=1, le=12)


class HomepageFeaturedConfig(BaseModel):
    title: str = "منتخب"
    subtitle: str = "پیشنهادهای هفته"
    catalog_label: str = "کاتالوگ ←"
    catalog_href: str = "/catalog"
    product_count: int = Field(default=6, ge=1, le=24)
    # latest = جدیدترین‌ها | manual = انتخاب دستی | category = یک دسته
    mode: str = Field(default="latest", pattern="^(latest|manual|category)$")
    product_ids: list[int] = Field(default_factory=list, max_length=48)
    category_id: int | None = None
    # سازگاری با تنظیمات قدیمی
    parent_slug: str | None = None


class HomepageSection(BaseModel):
    id: str
    enabled: bool = True


class HomepageConfig(BaseModel):
    sections: list[HomepageSection] = Field(
        default_factory=lambda: [
            HomepageSection(id="carousel"),
            HomepageSection(id="hero"),
            HomepageSection(id="featured"),
            HomepageSection(id="promo"),
        ]
    )
    hero: HomepageHeroConfig = Field(default_factory=HomepageHeroConfig)
    featured: HomepageFeaturedConfig = Field(default_factory=HomepageFeaturedConfig)
    show_promo_fallback: bool = True


class HomepageConfigPatch(BaseModel):
    sections: list[HomepageSection] | None = None
    hero: HomepageHeroConfig | None = None
    featured: HomepageFeaturedConfig | None = None
    show_promo_fallback: bool | None = None


class HomepagePublic(BaseModel):
    sections: list[HomepageSection]
    hero: HomepageHeroConfig
    featured: HomepageFeaturedConfig
    show_promo_fallback: bool
